src/java/org/apache/cassandra/index/sai/plan/VectorTopKProcessor.java

```diff
@@ -266,6 +290,49 @@ private PartitionResults processPartition(BaseRowIterator<?> partitionRowIterato
         return pr;
     }
 
+    /**
+     * Processes a single partition, calculating scores for rows and extracting tombstones.
+     */
+    private PartitionResults processPartition(BaseRowIterator<?> partitionRowIterator, ScoredPrimaryKey key, HashSet<PrimaryKey> updatedKeys) {
+        Row staticRow = partitionRowIterator.staticRow();
+        PartitionInfo partitionInfo = PartitionInfo.create(partitionRowIterator);
+        // VSTODO vector columns shouldn't be static, so can we remove this?
+        float keyAndStaticScore = getScoreForRow(key.partitionKey(), staticRow);
+        var pr = new PartitionResults(partitionInfo);
+
+        if (partitionRowIterator.hasNext())
+        {
+            Unfiltered unfiltered = partitionRowIterator.next();
+            // Always include tombstones for coordinator. It relies on ReadCommand#withMetricsRecording to throw
+            // TombstoneOverwhelmingException to prevent OOM.
+            if (unfiltered.isRangeTombstoneMarker())
+            {
+                pr.addTombstone(unfiltered);
+                return pr;
+            }
+
+            Row row = (Row) unfiltered;
+            float rowScore = getScoreForRow(null, row);
+            // Accept the Primary Key if its score, which comes from the vector index, is greater than the score
```

Comment:
Can we encapsulate the SK logic in retriever, something like 
if (retriever.shouldIncludeRow(key, row))
  pr.addRow(...)